builder/builder.go

```diff
@@ -522,137 +522,135 @@ func (b *Builder) BuildChroots(template *x509.Certificate, privkey *rsa.PrivateK
+		err = b.SignManifestMoM()
 		if err != nil {
-			helpers.PrintError(err)
+			return err
 		}
+		fmt.Println("Signed Manifest.MoM")
 	}
-}
 
-// BuildUpdate will produce an update consumable by the swupd client
-func (b *Builder) BuildUpdate(prefixflag string, minvflag int, formatflag string, signflag bool, publishflag bool, keepchrootsflag bool) error {
-	if formatflag != "" {
-		b.Format = formatflag
+	//
+	// Create full files.
+	//
+	fullfilecmd := exec.Command(prefixflag+"swupd_make_fullfiles", "-S", b.Statedir, b.Mixver)
+	fullfilecmd.Stdout = os.Stdout
+	fullfilecmd.Stderr = os.Stderr
+	err = fullfilecmd.Run()
+	if err != nil {
+		return errors.Wrapf(err, "couldn't create fullfiles")
 	}
 
-	if _, err := os.Stat(b.Statedir + "/www/version/format" + b.Format); os.IsNotExist(err) {
-		if err = os.MkdirAll(b.Statedir+"/www/version/format"+b.Format, 0777); err != nil {
-			helpers.PrintError(err)
-			return err
-		}
+	//
+	// Create zero packs.
+	//
+	zeropackArgs := []string{"--to", b.Mixver, "-S", b.Statedir}
+	if prefixflag != "" {
+		zeropackArgs = append(zeropackArgs, "--repodir", prefixflag)
 	}
-
-	// Step 1: create update content for the current mix
-	updatecmd := exec.Command(prefixflag+"swupd_create_update", "-S", b.Statedir, "--minversion", strconv.Itoa(minvflag), "-F", b.Format, "--osversion", b.Mixver)
-	updatecmd.Stdout = os.Stdout
-	updatecmd.Stderr = os.Stderr
-	err := updatecmd.Run()
+	zeropackcmd := exec.Command("mixer-pack-maker.sh", zeropackArgs...)
+	zeropackcmd.Stdout = os.Stdout
+	zeropackcmd.Stderr = os.Stderr
+	err = zeropackcmd.Run()
 	if err != nil {
-		helpers.PrintError(err)
-		return err
+		return errors.Wrapf(err, "couldn't create zero packs")
 	}
 
-	// We only need the full chroot from this point on, so cleanup the others to save space
-	if keepchrootsflag == false {
-		b.CleanChroots()
+	//
+	// Hardlink the duplicate files. This helps when keeping the bundle chroots, so
+	// save space from various chroots.
+	//
+	hardlinkcmd := exec.Command("hardlink", "-f", b.Statedir+"/image/"+b.Mixver+"/")
+	hardlinkcmd.Stdout = os.Stdout
+	hardlinkcmd.Stderr = os.Stderr
+	err = hardlinkcmd.Run()
+	if err != nil {
+		return errors.Wrapf(err, "couldn't perform hardlink step")
 	}
 
-	// Step 1.5: sign the Manifest.MoM that was just created
-	if signflag == false {
-		err = b.SignManifestMoM()
+	//
+	// Save upstream information.
+	//
+	if b.Upstreamurl != "" {
+		fmt.Printf("Saving the upstream URL: %s\n", b.Upstreamurl)
+		upstreamURLFile := filepath.Join(b.Statedir, "www", b.Mixver, "/upstreamurl")
+		err = ioutil.WriteFile(upstreamURLFile, []byte(b.Upstreamurl), 0644)
 		if err != nil {
-			helpers.PrintError(err)
-			return err
+			return errors.Wrapf(err, "couldn't write upstreamurl file")
+		}
+		fmt.Printf("Saving the upstream version: %s\n", b.Clearver)
+		upstreamVerFile := filepath.Join(b.Statedir, "www", b.Mixver, "upstreamver")
+		err = ioutil.WriteFile(upstreamVerFile, []byte(b.Clearver), 0644)
```

Comment:
I think this is fine, @mdhorn you said this was OK to move to a different part of the code as well right?